.github/workflows/main.yml

```diff
@@ -50,7 +53,33 @@ jobs:
           flake8 . --exit-zero --max-complexity=6
 
       - name: Upload python artifacts
+        if: ${{ github.ref_name }} == 'main'
         uses: actions/upload-artifact@v4
         with:
-          name: python-artifacts
+          name: ${{ env.PYTHON_ARTIFACTS_NAME }}
           path: .
+
+  docker-ci:
+    runs-on: ubuntu-latest
+    needs: python-ci
+    if: ${{ github.ref_name }} == 'main'
```

Comment:
Similar to the previous comment, ensure that the `if` condition correctly evaluates the branch name. Double-check the context in which `github.ref_name` is used to avoid any logical errors.